src/util.h

```diff
@@ -382,4 +382,49 @@ debug_hex_dump_slice(platform_log_handle *, uint64 grouping, slice data);
     : ((intval) < 1000) ? "3d"                                                 \
                         : "4d")
 
+// Size of on-stack char message buffer (mostly used in diagnostics)
+#define DIAG_MSGBUF_SIZE 80
+
+// Convenience specifier for output buffer to convert size with unit specifier
+#define SIZE_TO_STR_LEN 20
+
+// Min size of buffer needed if size is enclosed in (), as : '(~3.50 GiB)'
+#define SIZE_TO_STR_MINLEN (SIZE_TO_STR_LEN - 2)
+
+// Format a size value with unit-specifiers, in an output buffer.
+char *
+size_to_str(char *outbuf, size_t outbuflen, size_t size);
+
+char *
+size_to_fmtstr(char *outbuf, size_t outbuflen, const char *fmtstr, size_t size);
+
+/*
+ * Convenience caller macros to convert 'sz' bytes to return a string,
+ * formatting the input size as human-readable value with unit-specifiers.
+ *
+ * NOTE: These macros are best used when you have just one instance of a size
+ * to print in a function. If in places, e.g. diagnostic code, you have to
+ * print multiple size values across different statements, it's better to
+ * re-use on-stack output buffers.
+ */
+// char *size_str(size_t sz)
```

Comment:
One thing I noticed in my testing is that the name of the temporary, "b", is just too short and likely to collide with local variables.  Give it some long crazy name.

Reply:
Renamed this to `onstack_chartmp`.

(Also, I'd like to learn from you offline how you figured out that there could be likely name collision. I thought `{ }` scoping would prevent that.)